tests/typing/test_return_type.py

```
@@ -25,10 +26,11 @@ def _get_classes_from_module(module: ModuleType) -> list[type]:
             try:
                 issubclass(module_attr, object)
             except TypeError:
-                pass  # not a class
-            else:
-                if hasattr(module_attr, attr):
-                    class_types.append(module_attr)
+                continue  # not a class
+            if issubclass(module_attr, (_vtk.vtkAxesActor, _vtk.vtkTextActor)):
+                return  # Skip these classes
+            if hasattr(module_attr, attr):
+                class_types.append(module_attr)
```

@user27182, this was proving to be troublesome this would now yield what I might consider false positives. For example, `vtkAxesActor` and `vtkTextActor` both have `GetBounds()` and `GetCenter()` methods that return `None` and for that reason we don't wrap these methods in PyVista. However with VTK 9.4's new properties, `.bounds` and `.center` get mapped to thee methods and the test here is not valid for those wrappings. Do you think this is an appropriate workaround?

---

I'm not sure I understand. What is the desired or expected behavior?



> `vtkAxesActor` and `vtkTextActor` both have `GetBounds()` and `GetCenter()` methods that return `None`

is this vtk 9.4 behavior? on 9.3 I get:
``` python
import vtk 
vtk.vtkAxesActor().GetBounds()
# (-1.0, 1.0, -1.0, 1.0, -1.0, 1.0)
```

But in pyvista we don't define `bounds`

``` python
import pyvista as pv
pv.AxesActor().bounds
# AttributeError: 'AxesActor' object has no attribute 'bounds'
```

Without any knowledge of the failures you're seeing with tests or an understanding of the new vtk 9.4 behavior, it seems to me that we should probably define a `AxesActor.bounds` property, or am I missing something?

---

```py
>>> import vtk
>>> vtk.vtkTextActor().GetBounds()
>>> vtk.vtkAxesActor().GetBounds()
(-1.0, 1.0, -1.0, 1.0, -1.0, 1.0)
>>> 
```

I mistakenly only checked `vtkTextActor` and thought `vtkAxesActor` was also returning None but I may be getting different failures there.

> But in pyvista we don't define bounds

Correct, but VTK 9.4 implements a new wrapping mechanism that creates new properties like `bounds` which wrap `GetBounds()`. So even though we don't implement `bounds` it is now a property in VTK 9.4 and picked up by this dynamic parameterization. I'm going to just implement `bounds` and `center` properties for our wrapping of `vtkAxesActor` but this will still be an issue for `vtkTextActor`

---

Okay ignore `vtkAxesActor`. That is now fixed by implementing our own `bounds` and `center` properties.

---

I see. So there is now a confusing (and unavoidable?) blending of `vtk`'s API with our API. Our API previously promised that `bounds` returns a tuple (else raise `AttributeError`), but now it seems there is a risk that the return type for some objects is `tuple | None` if we haven't explicitly defined it ourselves...

Maybe we should override `vtkTextActor.bounds` to raise `NotImplementedError` instead of returning None?

---

> So there is now a confusing (and unavoidable?) blending of vtk's API with our API.

Yes, except if we override these properties then ours will always be used (well after this PR since I disable VTK's override mechanism in this PR). Most of this was handled in my original 9.4 compatibility work to nail down the method resolution order in #5650

> ... if we haven't explicitly defined it ourselves...
exaclty, thats the key here.

> Or maybe return uninitialized bounds like (1.0, -1.0, 1.0, -1.0, 1.0, -1.0) where min > max?

I think this is the most appropriate.